docs/proposals/default-service-plans.md

```diff
@@ -0,0 +1,544 @@
+1.  Replace any defaults set on the original class.
+
+##### Override the defaults for a broker level class
+
+1.  Define a new class (above).
+1.  Blacklist the original class so that it cannot be used anymore. _This relies upon functionality that is still in progress._
+
+##### Select a default plan for a service type
+
+```console
+$ svcat set-default plan NAME
+OLD_NAME is no longer the default plan for TYPE
+NAME is the default plan for TYPE
+
+```
+
+1.  If an existing plan is already the default for the type represented by the specified plan, the annotation marking the existing plan as the default is removed.
+1.  An annotation is added to the specified plan, marking it as the default for its type.
+
+### Developer
+
+
+#### Service Class Discovery
+
+Defaults are indicated with `*` after the type.
+
+
+```console
+$ svcat get classes
+	TYPE     	NAME           	DESCRIPTION          	SCOPE
++------------+------------------+---------------------+-----------+
+  redis        azure-redis        Azure Redis       	broker (osba)
+  postgres     azure-postgresql   Azure PostgreSQL  	broker (osba)
+  postgres     postgres-dev       Dev DB   	           cluster
+
+$ svcat get classes --type redis
+	TYPE     	NAME           	DESCRIPTION          	SCOPE
++------------+------------------+---------------------+-----------+
+  redis        azure-redis    	 Azure Redis       	broker (osba)
+
+$ svcat get plans --default
+	TYPE     	NAME     CLASS DESCRIPTION          	SCOPE
++------------+--------+--------------+---------------+-----------+
+  redis*     basic    azure-redis    Azure Redis     broker (osba)
+  postgres*  tiny     postgres-dev   Tiny Dev DB   	cluster
+```
+
+#### Legacy Provisioning
+
+This is unchanged. Create a service instance using a class+plan and define every parameter.
+
+Note: The parameters defined on the instance are merged with the parameters defined on the class, but since every value is set by the instance, all class-level defaults are overridden.
+
+
+#### Provisioning with Defaults
+
+Create a service instance using a class+plan. Individual parameters may be set on the instance to override a class-level default. Any unspecified parameters are defaulted using the values defined on the class.
+
+
+```console
+$ svcat provision mydb --class azure-mysql --plan basic50 \
+    --param location=westus
+```
+
+```yaml
+apiVersion: servicecatalog.k8s.io/v1beta1
+kind: ServiceInstance
+metadata:
+  name: mydb
+spec:
+  clusterServiceClassExternalName: azure-mysql
+  clusterServicePlanExternalName: basic50
+  parameters:
+    location: westus
+```
+
+
+The requested service instance is combined with the defaults defined on the class:
+
+
+```yaml
+apiVersion: servicecatalog.k8s.io/v1beta1
+kind: ClusterServiceClass
+metadata:
+  name: "997b8372-8dac-40ac-ae65-758b4a5075a5"
+spec:
+  externalID: "997b8372-8dac-40ac-ae65-758b4a5075a5"
+  externalName: azure-mysql
+  defaultProvisionParameters:
```

Comment:
I think it makes more sense to move this to the ServicePlan, or is there an assumption that a non default plan is not allowed default parameters? If the plan parameters do not match and the cluster operator is allowed to switch the default flag, it seems there could be a messy situation that could result

Reply:
Agreed, when I wrote this, I assumed that all plans in a class would use the same defaults (which turns out to be wrong 😉 ).